src/common/types/types.cpp

```diff
@@ -472,12 +452,36 @@ bool LogicalTypeUtils::isNumerical(const kuzu::common::LogicalType& dataType) {
     case LogicalTypeID::INT16:
     case LogicalTypeID::DOUBLE:
     case LogicalTypeID::FLOAT:
+    case LogicalTypeID::SERIAL:
         return true;
     default:
         return false;
     }
 }
 
+std::vector<LogicalType> LogicalTypeUtils::getAllValidComparableLogicalTypes() {
+    return std::vector<LogicalType>{LogicalType{LogicalTypeID::BOOL},
+        LogicalType{LogicalTypeID::INT64}, LogicalType{LogicalTypeID::INT32},
+        LogicalType{LogicalTypeID::INT16}, LogicalType{LogicalTypeID::DOUBLE},
+        LogicalType{LogicalTypeID::FLOAT}, LogicalType{LogicalTypeID::DATE},
+        LogicalType{LogicalTypeID::TIMESTAMP}, LogicalType{LogicalTypeID::INTERVAL},
+        LogicalType{LogicalTypeID::STRING}, LogicalType{LogicalTypeID::SERIAL}};
+}
+
+std::vector<LogicalTypeID> LogicalTypeUtils::getNumericalLogicalTypeIDs() {
+    return std::vector<LogicalTypeID>{LogicalTypeID::INT64, LogicalTypeID::INT32,
+        LogicalTypeID::INT16, LogicalTypeID::DOUBLE, LogicalTypeID::FLOAT, LogicalTypeID::SERIAL};
+}
+
+std::vector<LogicalTypeID> LogicalTypeUtils::getAllValidLogicTypeIDs() {
```

Comment:
Modify TODO to also include struct. Also this function seems to only be used in aggregate so we might need to remove it eventually.